framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeCopier.java

```diff
@@ -293,7 +293,20 @@ public AnnotatedTypeMirror visitTypeVariable(
     public AnnotatedTypeMirror visitPrimitive(
             AnnotatedPrimitiveType original,
             IdentityHashMap<AnnotatedTypeMirror, AnnotatedTypeMirror> originalToCopy) {
-        return makeCopy(original);
+        if (originalToCopy.containsKey(original)) {
+            return originalToCopy.get(original);
+        }
+
+        final AnnotatedPrimitiveType copy =
```

Comment:
Isn't this just inlining the body of `makeCopy`?

Reply:
`makeCopy` doesn't modify the `originalToCopy`.

Reply:
Yes, but why do you not call `makeCopy` instead of inlining it?
Why do we need to modify `originalToCopy` for primitive types? That is only needed to break cycles for type variables, which isn't a concern here, is it?